sql/src/main/java/io/crate/analyze/expressions/ExpressionAnalyzer.java

```diff
@@ -431,13 +432,34 @@ protected Symbol visitCast(Cast node, ExpressionAnalysisContext context) {
         }
 
         @Override
+        protected Symbol visitTryCast(TryCast node, ExpressionAnalysisContext context) {
+            DataType returnType = DATA_TYPE_ANALYZER.process(node.getType(), null);
+            Symbol sourceSymbol = process(node.getExpression(), context);
```

Comment:
isn't it possible to call `cast(sourceSymbol, targetType, context)` here and just wrap that + catch exception?